proxy/http/HttpConfig.cc

```diff
@@ -900,6 +900,59 @@ register_stat_callbacks()
                      (int)http_sm_finish_time_stat, RecRawStatSyncSum);
 }
 
+static bool
+set_negative_caching_list(const char *name, RecDataT dtype, RecData data, HttpConfigParams *c, bool update)
+{
+  bool ret = false;
+  HttpStatusBitset set;
+  // values from proxy.config.http.negative_caching_list
+  if (0 == strcasecmp("proxy.config.http.negative_caching_list", name) && RECD_STRING == dtype && data.rec_string) {
+    // parse the list of status code
+    ts::TextView status_list(data.rec_string, strlen(data.rec_string));
+    while (!status_list.empty()) {
+      auto is_sep{[](char c) { return isspace(c) || ',' == c || ';' == c; }};
+      ts::TextView span, token{status_list.ltrim_if(is_sep).take_prefix_if(is_sep)};
+      auto n = ts::svtoi(token, &span);
+      if (span.size() != token.size()) {
+        Error("Invalid status code '%.*s' for negative caching: not a number", static_cast<int>(token.size()), token.data());
+      } else if (n <= 0 || n >= 600) {
+        Error("Invalid status code '%.*s' for negative caching: out of range", static_cast<int>(token.size()), token.data());
+      } else {
+        set[n] = 1;
+      }
+    }
+  }
+  // set the return value
```

Comment:
It would probably be faster to do
```
if ((set ^ c->negative_caching_list).any()) {
  c->negative_caching_list = set;
  ret = ret || update;
}
```

Reply:
Good idea.

Reply:
I missed it the first time, but double checking there is an equality operator for `bitset`. I thought it was a bit odd to not have that.